rasterio/__init__.py

```diff
@@ -33,14 +33,14 @@
         DatasetReader, get_writer_for_path, get_writer_for_driver, MemoryFile)
     from rasterio.profiles import default_gtiff_profile
     from rasterio.transform import Affine, guard_transform
-    from rasterio.path import parse_path
+    from rasterio._path import _parse_path
```

Comment:
FYI @sgillies, this was a breaking API change. Though `from rasterio.path import parse_path` still works, `from rasterio import parse_path` used to work as well, but now doesn't.

stackstac at least is using `rasterio.parse_path` via the top-level API, so this has broken all stackstac versions. We're fixing it https://github.com/gjoseph92/stackstac/pull/155, but it means pervious stackstac versions will still try in install 1.3.0, but won't work with it.

It seems like a small number of projects may have been using this:
* https://github.com/search?q=rasterio.parse_path&type=Code
* https://github.com/search?q=rio.parse_path&type=Code

Reply:
Resolved in 5d5d946f. Thanks for catching this @gjoseph92 !